R/col2hex.R

```diff
@@ -0,0 +1,67 @@
+#' @title Get hex color
+#'
+#' @author Dania Machlab
+#' 
+#' @description The function returns a color in hex form given a valid name of a color in R.
+#' 
+#' @param col a \code{charachter} or \code{charachter} vector containing the names of the colors.
+#' @param alpha a numerical value in the range [0,1] or [0,255] that indicates the 
+#'   transparency of the color(s). If the given values are between 0 and 1, they 
+#'   are mapped to be between 0 and 255. An alpha value of 1 assumes the [0,1] range
+#'   and provides maximum color. The default is set to 255.
+#'   
+#' @return a \code{charachter} or \code{charachter} vector with the hex colors. 
+#' 
+#' @examples 
+#' y <- rnorm(1000,0,1)
+#' cols <- rep("red", length(y))
+#' alpha <- seq(0,1,length.out=length(y))
+#' hexcols <- col2hex(cols, alpha)
+#' plot(1:length(y), y, bg=hexcols, pch=21)
+#' 
+#' y <- rnorm(1000,0,1)
+#' cols <- rep("red", length(y))
+#' alpha <- seq(0,255,length.out=length(y))
+#' hexcols <- col2hex(cols, alpha)
+#' plot(1:length(y), y, bg=hexcols, pch=21)
+#' 
+#' @importFrom grDevices col2rgb rgb
+#'
+#' @export
+col2hex <- function(col, alpha=255) {
+  
+  # checks
+  if(!is.character(col)){
```

Comment:
I think the check `is.character(col)` is fine but not absolutely needed. In R, colors can also be specified as integers (for an integer value, it will convert it using `col2rgb`, which you are anyway using below). Also, `.isValidColor` knows about that and checks for it.